build/linter/constructor/analyzer.go

```diff
@@ -67,7 +67,12 @@ func getConstructorList(t types.Type) []string {
 		if !ok {
 			continue
 		}
-		if named.Obj().Name() == "Constructor" && named.Obj().Pkg().Path() == "github.com/pingcap/tidb/util/linter/constructor" {
+		if ignoreFields != nil {
+			if _, ok := ignoreFields[field.Name()]; ok {
```

Comment:
should we don't recursively check the fields in line 90 instead of adding `ignoreFields` argument? Not sure `(*inspector.Inspector).WithStack` still can do a recursively check.

Reply:
`(*inspector.Inspector).WithStack` will do the recursively check. I've added the test case:

```
	// specified field with manually constructed struct is not allowed
	_ = compositeImplicitInitiate1{
		StructWithSpecificConstructor: StructWithSpecificConstructor{}, // want `struct can only be constructed in constructors NewStructWithSpecificConstructor, AnotherConstructor`
	}
```

Even though the `StructWithSpecificConstructor` is ignored, constructing `StructWithSpecificConstructor{}` is still not allowed.